src/main/webapp/app/exam/manage/exams/exam-checklist-component/exam-edit-workingtime-dialog/exam-edit-working-time-dialog.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component } from '@angular/core';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { faBan, faCheck, faSpinner } from '@fortawesome/free-solid-svg-icons';
+import { Exam } from 'app/entities/exam.model';
+import { EntityResponseType, ExamManagementService } from 'app/exam/manage/exam-management.service';
+
+@Component({
+    selector: 'jhi-edit-working-time-dialog',
+    templateUrl: './exam-edit-working-time-dialog.component.html',
+})
+export class ExamEditWorkingTimeDialogComponent {
+    exam: Exam;
+    examChange?: (exam: Exam) => void; // somehow, event emitter does not work
```

Comment:
Can you elaborate on this? Technically it should work if you pass the event emitter through

Reply:
I know, tried it and messed around for over an hour and digging down into the source. It was like that:

```ts
    @Input() exam: Exam;
    @Output() examChange = new EventEmitter<Exam>();

    // ...

    confirmUpdateWorkingTime(): void {
        if (!this.isWorkingTimeValid()) return;
        this.isLoading = true;
        this.examManagementService.updateWorkingTime(this.exam.course!.id!, this.exam.id!, this.workingTimeSeconds).subscribe({
            next: (res: HttpResponse<Exam>) => {
                this.isLoading = false;
                res.body && this.examChange.emit(res.body);
                this.clear();
            },
            error: () => {
                this.isLoading = false;
                // TODO: error handling
            },
        });
    }
```

This gave me the following exception:

```
eventemitter this._throwIfClosed is not a function
```

For sake of speed, I ditched the event emitter for now, and used a good old and ugly callback. Please help me if I miss sth. obvious. Somehow, the `this` might be bound wrong, but I don't see how.

Reply:
Fixed it.